scripts/check-redact-log

```diff
@@ -1,9 +1,21 @@
 #!/usr/bin/env bash
 # This script checks if there is plain (not redacted) log in TiKV.
+set -euo pipefail
 
-if grep -r -n -P '(?<!hex_)encode_upper' -C 1 --color=always --include \*.rs --exclude hex.rs --exclude-dir target . ; then
-    echo "To print user data into info logs or error messages, use log_wrappers::Value() instead of hex::encode_upper. The former will respect \`security.redact-info-log\` config and filter out user data from info log if needed. Otherwise, use \`log_wrappers::hex_encode_upper\` to get around the lint error. See https://github.com/tikv/tikv/pull/9250 for more information." 
-    exit 1
+function error_msg() {
+        echo "To print user data into info logs or error messages, use log_wrappers::Value() instead of hex::encode_upper. The former will respect \`security.redact-info-log\` config and filter out user data from info log if needed. Otherwise, use \`log_wrappers::hex_encode_upper\` to get around the lint error. See https://github.com/tikv/tikv/pull/9250 for more information." >&2
+}
+
+if command -v uname >/dev/null && [[ "$(uname)" == "Darwin" ]] ; then
+   if grep -r -n --color=always --include '*.rs' --exclude hex.rs --exclude-dir target 'encode_upper' . | grep -v log_wrappers ; then
```

Comment:
Probably just using the same command is enough for all platforms.

Reply:
Why is it marked resolved? I can see different commands are still used for different platforms.

Reply:
On macOS, `grep` doesn't support look back pattern (`{except hex_}encode_upper`). If we use grep once and then `grep -v`, the output will lose context information `-C`, which shows adjacent lines around `encode_upper` and is friendly to developers. Therefore, to retain context information correctly, we need to use one grep with look back pattern, and have to use two separate commands for different platforms. The result is that, on macOS, no context information is shown, and on Linux, adjacent lines will be shown.

Reply:
Get it, thanks!